src/som/vmobjects/SFileDescriptor.java

```diff
@@ -171,17 +172,7 @@ public void setBufferSize(final int bufferSize) {
   }
 
   public void setMode(final SSymbol mode) {
-    try {
-      this.accessMode = AccessModes.valueOf(mode.getString());
-    } catch (IllegalArgumentException e) {
-      signalInvalidAccessMode(mode.getString());
-    }
-  }
-
-  public static void signalInvalidAccessMode(final Object mode) {
-    CompilerAsserts.neverPartOfCompilation();
-    KernelObj.signalException("signalArgumentError:", "File access mode invalid, was: "
-        + Objects.toString(mode) + " " + AccessModes.VALID_MODES);
+    this.accessMode = AccessModes.valueOf(mode.getString());
```

Comment:
I don't get this change. Where did the exception go?

Reply:
In the sender, so it can have the exceptionSignalNode.